File: Geek Assistant (Android AIO)/Forms/ToU.vb
Imports System.IO

Public Class ToU
    Public Shared RunningAlready As Boolean = False

    Private Sub TermsOfUse_Closing(sender As Object, e As EventArgs) Handles MyBase.Closing
        My.Settings.Save()
        RunningAlready = False
        If ToURead_Timer.Enabled Then ToURead_Timer.Stop()

    End Sub

    Public ToURead_TimeAmount As Integer
    Private Sub TermsOfUse_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        GA_SetTheme.Run(Name)

        Copyright.Text = GA_Ver.Run("ToU")

        'TermsOfUse.rtf 
        Dim Current_ToUrtf As String
        If Not Directory.Exists(GA) Then Directory.CreateDirectory(GA)
        If My.Settings.DarkTheme Then
            Current_ToUrtf = $"{GA}\TermsOfUse-Dark.rtf"
            If File.Exists(Current_ToUrtf) Then File.Delete(Current_ToUrtf)
            File.WriteAllText(Current_ToUrtf, My.Resources.TermsOfUse_Dark)
        Else
            Current_ToUrtf = $"{GA}\TermsOfUse.rtf"
            If File.Exists(Current_ToUrtf) Then File.Delete(Current_ToUrtf)
            File.WriteAllText(Current_ToUrtf, My.Resources.TermsOfUse)
        End If
        With TermsOfUse_Box
            .SelectionLength = 0
            .LoadFile(Current_ToUrtf)
        End With
        ToU_Reject.Select()


        'Accept || xy // Size xy: 337, 405 // 173, 46
        'Reject || xy // Size xy: 204, 405 // 133, 46
        If RunningAlready Then
            SetBounds((Main.Width / 2) - (Width / 2) + Main.Location.X, Main.Top, Width, Height)
            TermsOfUse_Box.Height = Height - 202
            AcceptCheck_ToU.Visible = False
            'With DontSkip_Label
            '    .Left = 363
            '    .ForeColor = Color.DimGray
            '    .Text = "Terms of use"
            'End With
            DontShow_ToU.Visible = False
            With ToU_Reject
                .Text = "✖"
                .Left = 394
                .Width = 58
            End With
            With ToU_Accept
                .Left = 452
                .Width = 58
                .Enabled = True
                .BackColor = Color.White
                .Text = "✔"
                .Font = New Font("Segoe UI", 13)
            End With
            'With Copyright
            '    .Left = 26
            '    .Top = Height - 84
            '    .Font = New Font("Segoe UI", 12, FontStyle.Regular)
            '    .ForeColor = Color.FromArgb(64, 64, 64)
            '    .BackColor = ButtonsBG_UI.BackColor
            'End With
            ToU_Accept.Select()
            Exit Sub
        End If

        If Version.Revision < 3 Then
            ToURead_Timer.Start()
        Else ''Skip the 6sec timer for #Dev version
            ToURead_TimeAmount = 6
            AcceptCheck_ToU.Checked = True
        End If

        If My.Settings.ToU_dontShow Then  'ToU:[ Dont show ToU >> Show Startup:[ Don't show AppMode >> Start newbie/moderate ] ]
            If My.Settings.AppMode_dontshow Then Main.Show() Else AppMode.Show()
            Close()
        End If
    End Sub
    Private Sub ToU_Reject_MouseEnter_MouseDown_KeyDown(sender As Object, e As EventArgs) Handles ToU_Reject.MouseEnter, ToU_Reject.MouseDown, ToU_Reject.KeyDown
        ToU_Reject.ForeColor = Current_bgColor()
    End Sub
    Private Sub ToU_Reject_MouseLeave_KeyUp(sender As Object, e As EventArgs) Handles ToU_Reject.MouseLeave, ToU_Reject.KeyUp
        ToU_Reject.ForeColor = Current_fgColor()
    End Sub
    Private Sub ToU_Reject_Click(sender As Object, e As EventArgs) Handles ToU_Reject.Click

        '>DEPRECATED
        'If RunningAlready Then
        '    Close()
        '    Exit Sub
        'End If
        '<DEPRECATED

        ErrorCodeTrack("ToU-R-H") ' ToU Rejected Hide
        GA_HideAllForms.Run(True, Nothing)
        Dim RejectTou_MessageBox = GA_infoAsk.Run("Rejected the terms of use", "Sorry for any inconvenience. You could contact the developer for further discussion.", "Exit Geek Assistant", "Return")
        If RejectTou_MessageBox Then
            Application.Exit()
        Else
            ErrorCodeTrack("ToU-R-S")
            GA_HideAllForms.Run(False, Me)
        End If
    End Sub

    Private Sub ToU_Accept_MouseEnter_MouseDown_KeyDown(sender As Object, e As EventArgs) Handles ToU_Accept.MouseEnter, ToU_Accept.MouseDown, ToU_Accept.KeyDown
        ToU_Accept.ForeColor = Current_bgColor()
    End Sub
    Private Sub ToU_Accept_MouseLeave_KeyUp(sender As Object, e As EventArgs) Handles ToU_Accept.MouseLeave, ToU_Accept.KeyUp
        ToU_Accept.ForeColor = Current_fgColor()
    End Sub
    Private Sub ToU_Accept_Click(sender As Object, e As EventArgs) Handles ToU_Accept.Click
        If DontShow_ToU.Checked Then My.Settings.ToU_dontShow = True
        If RunningAlready Then
            Close()
            Exit Sub
        End If
        If My.Settings.AppMode_dontshow Then Main.Show() Else AppMode.Show()
        Close()
    End Sub

    Private Sub AcceptCheck_ToU_CheckedChanged(sender As Object, e As EventArgs) Handles AcceptCheck_ToU.CheckedChanged
        If AcceptCheck_ToU.Checked And ToURead_TimeAmount = 6 Then
            With ToU_Accept
                .Enabled = True
                .BackColor = BackColor
            End With
        Else
            With ToU_Accept
                .Enabled = False
                .BackColor = ButtonsBG_UI.BackColor
            End With
        End If
    End Sub

    Private Sub ToURead_Timer_Tick(sender As Object, e As EventArgs) Handles ToURead_Timer.Tick
        If ToURead_TimeAmount = 5 Then
            ToURead_Timer.Stop()
        End If
        ToURead_TimeAmount += 1
        ToU_Accept.Text = "&ACCEPT (" & (6 - ToURead_TimeAmount).ToString & ")"
        If ToURead_TimeAmount = 6 Then
            ToU_Accept.Text = "&ACCEPT"
            If AcceptCheck_ToU.Checked Then
                With ToU_Accept
                    .Enabled = True
                    .BackColor = Color.White
                End With
            End If
        End If
    End Sub

    Private Sub DontShow_ToU_CheckedChanged(sender As Object, e As EventArgs) Handles DontShow_ToU.CheckedChanged

    End Sub
End Class

File: Geek Assistant (Android AIO)/Forms/ToU.cs
 
using System; 
using System.Drawing;
using System.IO; 
using System.Windows.Forms;

namespace GeekAssistant {
    public partial class ToU : Form {

        public ToU() {
            InitializeComponent();
        }

        public bool RunningAlready = false;
        protected override void OnFormClosing(FormClosingEventArgs e) {
            base.OnFormClosing(e);
            common.S.Save();
            RunningAlready = false;
            if (ToURead_Timer.Enabled) ToURead_Timer.Stop();
        }

        public int ToURead_TimeAmount = 0;
        private void ToU_Load(object sender, EventArgs e) {
            GA_SetTheme.Run(Name);

            Copyright_Label.Text = GA_Ver.Run("ToU");

            Load_ToU_rtf();

            ToU_Reject.Select();

            //Accept || xy // Size xy: 337, 405 // 173, 46
            //Reject || xy // Size xy: 204, 405 // 133, 46
            if (RunningAlready) {
                SetBounds((common.Home.Width / 2) - (Width / 2) + common.Home.Location.X, common.Home.Top, Width, Height);
                AcceptCheck_ToU.Visible = false;
                DontShow_ToU.Visible = false;
                ToU_Reject.Text = "✖";
                ToU_Reject.Left = 394;
                ToU_Reject.Width = 58;

                ToU_Accept.Left = 452;
                ToU_Accept.Width = 58;
                ToU_Accept.Enabled = true;
                //.BackColor = BackColor
                ToU_Accept.Text = "✔";
                ToU_Accept.Font = new Font("Segoe UI", 13);

                ToU_Accept.Select();
                return;
            }

            if (common.V.Revision < 3) {
                ToURead_Timer.Start();
            } else { ////Skip the 6sec timer for #Dev version 
                ToU_Accept.BackColor = BackColor;
                ToURead_TimeAmount = 6;
                AcceptCheck_ToU.Checked = true;
            }

            if (common.S.ToU_dontShow) {  //ToU:[ Dont show ToU >> Show Startup:[ Don//t show AppMode >> Start newbie / moderate ] ]
                if (common.S.AppMode_dontshow) common.Home.Show(); else common.AppMode.Show();
                Close();
            }
        }


        private void Load_ToU_rtf() {
            string terms_rtf = $"{common.GA}\terms.rtf"; //set destination
            var Current_rtf = prop.GA.TermsOfUse; //set default file
            if (common.S.DarkTheme) Current_rtf = prop.GA.TermsOfUse_Dark; //set dark file if dark theme
            if (!Directory.Exists(common.GA)) Directory.CreateDirectory(common.GA); //create GA folder if not present
            if (File.Exists(terms_rtf)) File.Delete(terms_rtf); //delete old file if present
            File.WriteAllText(terms_rtf, Current_rtf); //write file to GA
            TermsOfUse_Box.LoadFile(terms_rtf); //load file to the box
            TermsOfUse_Box.SelectionLength = 0; //deselect everything
        }

        private void ToU_Reject_MouseEnter_MouseDown_KeyDown(Object sender, EventArgs e) {
            ToU_Reject.ForeColor = GA_SetTheme.Current_bgColor();
        }

        private void ToU_Reject_MouseLeave_KeyUp(Object sender, EventArgs e) {
            ToU_Reject.ForeColor = GA_SetTheme.Current_fgColor();
        }

        private void ToU_Reject_Click(Object sender, EventArgs e) { 
            common.ErrorInfo.code = "ToU-R-H"; // ToU Rejected Hide
            GA_HideAllForms.Run(true);
            if (GA_infoAsk.Run("Rejected the terms of use",
                                 "Sorry for any inconvenience. You could contact the developer for further discussion.",
                               "Exit", "Back")) { 
                Application.Exit();
            } else {
                common.ErrorInfo.code = "ToU-R-S";
                GA_HideAllForms.Run(false);
            }
        }

        private void ToU_Accept_MouseEnter_MouseDown_KeyDown(Object sender, EventArgs e) {
            ToU_Accept.ForeColor = GA_SetTheme.Current_bgColor();
        }
        private void ToU_Accept_MouseLeave_KeyUp(Object sender, EventArgs e) {
            ToU_Accept.ForeColor = GA_SetTheme.Current_fgColor();
        }
        private void ToU_Accept_Click(Object sender, EventArgs e) {
            if (DontShow_ToU.Checked) {
                common.S.ToU_dontShow = true;
                if (RunningAlready) {
                    Close();
                    return;
                }
                if (common.S.AppMode_dontshow) { common.Home.Show(); } else {
                    common.AppMode.Show();
                    Close();
                }
            }
        }
        private void AcceptCheck_ToU_CheckedChanged(Object sender, EventArgs e) {
            if (AcceptCheck_ToU.Checked & ToURead_TimeAmount == 6) {

                ToU_Accept.Enabled = true;
                ToU_Accept.BackColor = BackColor;
            } else {
                ToU_Accept.Enabled = false;
                ToU_Accept.BackColor = ButtonsBG_UI.BackColor;
            }
        }

        private void ToURead_Timer_Tick(Object sender, EventArgs e) {

            ToU_Accept.BackColor = ButtonsBG_UI.BackColor;
            if (ToURead_TimeAmount == 5) {
                ToURead_Timer.Stop();
                ToURead_TimeAmount += 1;
                ToU_Accept.Text = $"&ACCEPT ({(6 - ToURead_TimeAmount)})";
                if (ToURead_TimeAmount == 6) {
                    ToU_Accept.Text = "&ACCEPT";
                    if (AcceptCheck_ToU.Checked) {
                        ToU_Accept.Enabled = true;
                        ToU_Accept.BackColor = BackColor;
                    }
                }
            }
        }
    }
}
